frontend/src/pages/ForgotPasswordPage.tsx

```diff
@@ -0,0 +1,251 @@
+import React, { useState, useEffect, FormEvent, useMemo } from 'react';
+import { Alert } from 'react-bootstrap';
+import { Trans, useTranslation } from 'react-i18next';
+import { Link } from 'react-router-dom';
+import classNames from 'classnames';
+import { ModalContent } from '../components/modals/modalparts/ModalContent';
+import { ModalHeader } from '../components/modals/modalparts/ModalHeader';
+import { Footer } from '../components/Footer';
+import { paths } from '../routers/paths';
+import { Input, errorState } from '../components/forms/FormField';
+import { LoadingSpinner } from '../components/LoadingSpinner';
+import { ReactComponent as MailIcon } from '../icons/mail_icon.svg';
+import { api } from '../api/api';
+import css from './ForgotPassword.module.scss';
+
+const classes = classNames.bind(css);
+
+export const ForgotPasswordPage = () => {
+  const { t } = useTranslation();
+  const defaultDisableTime = 30; // Seconds
+
+  const [linkSent, setLinkSent] = useState(false);
+  const [email, setEmail] = useState('');
+  const [sendDisabled, setSendDisabled] = useState(false);
+  const [sendDisableTime, setSendDisableTime] = useState(
+    defaultDisableTime / 2,
+  );
+  const [resendDisabled, setResendDisabled] = useState(false);
+  const [resendDisableTime, setResendDisableTime] = useState(
+    defaultDisableTime,
+  );
```

Comment:
I think the two flags & times should use just one and same flag & time. Just setting the time to different initial value from different places.

This would simplify the code and keep the timer consistent when clicking to "change address" after resending.

Reply:
This continues from the above explanation. The sendDisableTime can be going 15->14->13->12 etc. while the resendDisableTime is going from 30->29->... so they needed their own times in this particular implementation (at least I think they are needed).